core/transformation/pathfinders/round_1/control_mappings.py

```diff
@@ -104,3 +108,17 @@ def _intervention_theme_to_standard_outcomes(standard_outcomes_df: pd.DataFrame)
         for _, row in standard_outcomes_df.iterrows()
         if not pd.isna(row["Intervention theme"])
     }
+
+
+def _output_outcome_uoms(control_data_df: pd.DataFrame, column_name: str) -> dict[str, list[str]]:
+    """Creates a mapping from standard/bespoke outcomes or ouputs to their unit of measurement.
+
+    :param control_data_df: Dataframe of the extracted control data table
+    :param column_name: String value of the column name from the relevant control table to be used for the mapping
+    :return: Dictionary of output or outcome names to a list of their unit of measurement
+    """
+    return {
+        row[column_name]: control_data_df.loc[control_data_df[column_name] == row[column_name], "UoM"].tolist()
+        for _, row in control_data_df.iterrows()
+        if not pd.isna(row["UoM"])
+    }
```

Comment:
I can't see a unit test for this function, would it be possible to create one? Or am I missing something?

Reply:
Ah no you're right, it's not checked in the same way by the general unit test as it's not a direct table mapping - writing a test now!